jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -753,71 +801,96 @@ def decontaminate_image(scidata_bkg, tracemodels, subarray):
+    spec_list : list of SpecModel
+        If save_tiktests is True, returns a list of the model spectra for each Tikhonov factor tested,
+        with the best-fitting spectrum last in the list.
+        If save_tiktests is False, returns a one-element list with the best-fitting spectrum.
+    
+    Notes
+    -----
+    The last spectrum in the list of SpecModels lacks the "chi2", "chi2_soft_l1", "chi2_cauchy", and "reg"
+    attributes, as these are only calculated for the intermediate models. The last spectrum is not
+    necessarily identical to any of the spectra in the list, as it is reconstructed according to
+    mask_rebuild instead of fit respecting mask_fit.
+
+    # TODO are all of these behaviors for the last spec in the list the desired ones?
+    """
 
     # The throughput and kernel is not needed here; set them so they have no effect on the extraction.
     def throughput(wavelength):
         return np.ones_like(wavelength)
     kernel = np.array([1.])
-
-    # Set reference file arguments
     ref_file_args[2] = [throughput]
     ref_file_args[3] = [kernel]
 
-    # ###########################
-    # First, generate an estimate
-    # (only if the initial guess of tikhonov factor range is not given)
-    # ###########################
-
-    if tikfac_log_range is None:
-        # Initialize the engine
-        engine = ExtractionEngine(*ref_file_args,
-                                  wave_grid=wave_grid,
-                                  orders=[order],
-                                  mask_trace_profile=[mask_fit])
-
-        # Extract estimate
-        spec_estimate = engine.__call__(data=data_order, error=err_order)
-
-        # Interpolate
-        idx = np.isfinite(spec_estimate)
-        estimate_spl = UnivariateSpline(wave_grid[idx], spec_estimate[idx], k=3, s=0, ext=0)
-
-    # ##################################################
-    # Second, do the extraction to get the best estimate
-    # ##################################################
     # Define wavelength grid with oversampling of 3 (should be enough)
     wave_grid_os = oversample_grid(wave_grid, n_os=3)
-    wave_grid_os = wave_grid_os[wave_grid_os > 0.58]
+    wave_grid_os = wave_grid_os[wave_grid_os > ORDER2_SHORT_CUTOFF]
 
     # Initialize the Engine.
     engine = ExtractionEngine(*ref_file_args,
                               wave_grid=wave_grid_os,
-                              orders=[order],
-                              mask_trace_profile=[mask_fit])
+                              mask_trace_profile=[mask_fit],
+                              orders=[order],)
 
     # Find the tikhonov factor.
     # Initial pass with tikfac_range.
-    if tikfac_log_range is None:
-        guess_factor = engine.estimate_tikho_factors(estimate_spl)
-        log_guess = np.log10(guess_factor)
-        factors = np.log_range(log_guess - 2, log_guess + 8, 10)
-    else:
-        factors = np.logspace(tikfac_log_range[0], tikfac_log_range[-1] + 8, 10)
```

Comment:
It was very non-obvious to me why a random 8 orders of magnitude were being added to the tikfac_log_range here.  If those are the factors that are actually desired, why not just change the input tikfac_log_range?  So I removed that but want to make sure the current way the test factors are determined is okay.